### DISCUSSION TITLE
VirtualMachineError: revert: SafeERC20: low-level call failed

### DISCUSSION THREAD

**Discussion by 11296925** [smartcontractkit/full-blockchain-solidity-course-py#1579] [Q&A]
Hi, Currently I am working on lesson 10 where I try to borrow from Aave. I got stuck at the point to deposit weth into the lending pool address. My code seems to be correct, but whatever I try, I receive the following error:

<img width="575" alt="SafeERC20 Error2" src="https://user-images.githubusercontent.com/36366013/178267866-50f01b6c-2394-4b1a-b135-f35ed99448bb.png">

The error occurs when running the line:

    tx = lending_pool.deposit(
        erc20_address, amount, account.address, 0, {"from": account}
    )

Does anyone have an idea about how to fix this? Thanks!

**Comment by SJ50** [comment]
Are you approving contract before making a deposit? 
Approving amount should be equal or higher than deposit amount. If you trust the contract than approving amount can be max (2**256 -1).

**Reply by cromewar** [reply]
I think is is the best solution, does this worked @11296925 ?

**Reply by 11296925** [reply]
Thanks for the answers and apologies for my late response. I did all these things but it doesn't work. I do approve the contract before making a deposit with the same amount as the amount to deposit. This is my code:

<img width="428" alt="Aave_borrow code" src="https://user-images.githubusercontent.com/36366013/180283153-dc25246a-4328-4482-9d9c-ba3245bebac0.png">

I also tried adding extra waiting time in between the approval and the deposit function, and approving an amount 100 times higher than the amount to deposit. But this also didn't fix the error.

**Reply by cromewar** [reply]
I wonder wich version of the lending pool are you using? because I think the version in the video is outdated and the way it works now is different.
